src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPoolWorkQueue.cs

```diff
@@ -1392,8 +1392,6 @@ void IThreadPoolWorkItem.Execute()
                 // Reset thread state after all user code for the work item has completed
                 currentThread.ResetThreadPoolThread();
             }
-
-            ThreadInt64PersistentCounter.Add(tl.threadLocalCompletionCountObject!, completedCount);
```

Comment:
It should still count each of the IO completion callbacks that are called in batch above. `ThreadPoolWorkQueue.Dispatch` would also count the current work item (`ThreadPoolTypedWorkItemQueue`'s queue processing work item), which could be approximately discounted here. It could be something like:

```c#
// Discount a work item here to avoid counting most of the queue processing work items
if (completedCount > 1)
{
    ThreadInt64PersistentCounter.Add(tl.threadLocalCompletionCountObject!, completedCount - 1);
}
```